### PR TITLE
Update  AthenaJDBC from 1.x to 2.x

### LINKED ISSUE #2: Update AthenaJDBC version to 2.0.5
I want to use `AthenaJDBC41_2.0.5`. It is different from the configuration of `AthenaJDBC41-1.1.0`.  
[Migration from Previous Version of the JDBC Driver](https://docs.aws.amazon.com/athena/latest/ug/connect-with-jdbc.html)

[Installation and Configuration Guide](https://s3.amazonaws.com/athena-downloads/drivers/JDBC/SimbaAthenaJDBC_2.0.5/docs/Simba+Athena+JDBC+Driver+Install+and+Configuration+Guide.pdf)

### New feature in Version 2.0.5 
[Amazon Athena releases an updated JDBC driver with improved performance when retrieving results](https://docs.aws.amazon.com/ja_jp/athena/latest/ug/connect-with-jdbc.html)

### PR DESCRIPTION
**Pull request by matsuiyuki**
close #2 

Config Item name and order conformed to "Installation and Configuration Guide".
Note: "AwsRegion" is included in "Url".

Task
- [x] update Config
- [x] update README
- [x] fix readOnly Connection(on hold)

### CODE REVIEW

**Comment by matsuiyuki** [comment]
@zaneli Now that the source code has been completed, I would like you to review my code.

File: src/main/scala/scalikejdbc/athena/Config.scala [hunk 1]
```diff
@@ -64,21 +69,44 @@ class Config(dbName: Any) {
...
+
+  val S3OutputLocation = "S3OutputLocation"
+  val S3OutputLocationPrefix = "S3OutputLocationPrefix"
+
+  val AwsCredentialsProviderArguments = "AwsCredentialsProviderArguments"
+  val AwsCredentialsProviderClass = "AwsCredentialsProviderClass"
+  val BinaryColumnLength = "BinaryColumnLength"
+  val ComplexTypeColumnLength = "ComplexTypeColumnLength"
+  val ConnectionTest = "ConnectionTest"
+  val ConnectionTimeout = "ConnectTimeout"
```
Comment (zaneli) [inline src/main/scala/scalikejdbc/athena/Config.scala]: I think that it is better to match val name to config option name.
```scala
val ConnectTimeout = "ConnectTimeout"
```

File: src/main/scala/scalikejdbc/athena/Config.scala [hunk 2]
```diff
@@ -38,13 +41,15 @@ class Config(dbName: Any) {
 
   private[athena] lazy val url: String = map.getOrElse(Url, throw new ConfigException(s"no configuration setting: key=$prefix.$Url"))
 
+  private[athena] lazy val readOnly: Boolean = map.get(ReadOnly).fold(true) { !_.matches("^false$") }
```
Comment (zaneli) [inline src/main/scala/scalikejdbc/athena/Config.scala]: What do you think about change readOnly to optional configuration?

```scala
private[athena] lazy val readOnly: Option[Boolean] = map.get(ReadOnly).flatMap(v => Try(v.toBoolean).toOption)
```

```scala
val c = new AthenaConnection(DriverManager.getConnection(config.url, config.options))
config.readOnly.foreach(c.setReadOnly)
```

File: src/main/scala/scalikejdbc/athena/Config.scala (hunk 1 above)
Comment (matsuiyuki) [reply src/main/scala/scalikejdbc/athena/Config.scala]: Uh-oh. sorry. to correct

File: src/test/scala/scalikejdbc/athena/ConfigSpec.scala [hunk 3]
```diff
@@ -7,22 +7,25 @@ class ConfigSpec extends FunSpec with OptionValues {
...
     it("default db") {
       val config = new Config("default")
-      assert(config.url === "jdbc:awsathena://athena.us-east-2.amazonaws.com:443")
-      assert(config.options.getProperty("s3_staging_dir") === "s3://query-results-bucket/folder/")
-      assert(config.options.getProperty("log_path") === "logs/application.log")
+      assert(config.url === "jdbc:awsathena://AwsRegion=us-east-2")
+      assert(config.options.getProperty("S3OutputLocation") === "s3://query-results-bucket/folder/")
+      assert(config.options.getProperty("LogPath") === "logs/application.log")
       assert(config.getTmpStagingDir.isEmpty)
+      assert(config.readOnly === Some(false))
```
Comment (zaneli) [inline src/test/scala/scalikejdbc/athena/ConfigSpec.scala]: In the same way as `getTmpStagingDir` assertion,
please use `config.readOnly.value`, `config.readOnly.isEmpty`.

**Comment by zaneli** [comment]
@matsuiyuki 
Please fix https://github.com/zaneli/scalikejdbc-athena/pull/3#discussion_r236172284 and organize commits, and I will merge.

**Review by zaneli — APPROVED** [review]
LGTM 👍

**Comment by zaneli** [comment]
@matsuiyuki
Thank you!
I'll release scalikejdbc-athena 0.2.0.

**Pull request merged**